### ISSUE TITLE
Optimize archive import with lots of dead (offline pods) contacts

### ISSUE THREAD

**Issue by jhass** [diaspora/diaspora#8019]
So my account migration took about 36 hours or so, that is because it tries to fetch all dead people in my contacts and reshares. If we can somehow  cache that a fetch failed from the validator pass, or just disable the fetch in the import pass, that should provide a tremendous speedup for the case of many dead contacts.

**Comment by goobertron** [comment]
Would it make sense to prune these from the account data before it is packaged for export?

**Comment by SuperTux88** [comment]
No, this needs to be handled during the import. Pods are maybe still up when you export your profile, but are down a week later, or if your export is really old, maybe even years later (and the import works also with an old archive, every archive created with 0.7.0.0 or newer should work for import).

**Comment by goobertron** [comment]
I understand that in some cases an archive will be imported long after export, but I don't think that was the case in Jonne's test. I imagined that in this case it was a lot of long-closed accounts that had been sharing with his account and had never been pruned before.

If that's the case, or if that's a possible case, it might be worth either periodically pruning accounts from dead pods from an account's active sharing relationships (perhaps putting them in an 'inactive' status so they can be re-activated as contacts if the account becomes live again) or doing this pruning at the time of export as well as at import. There then might be a cleaner exported data package, and neither step would take a huge amount of time.

If I'm misunderstanding some technical detail, don't feel you need to spend time trying to explain it to me; just ignore me. I thought I'd make the suggestion in case I had thought of a useful method to think about. If not, that's fine.

**Comment by SuperTux88** [comment]
You need to do it during import anyway, so there is no need to do it twice, and the export is also already a long enough process anyway. And there is the possibility that your new pod already knows the (then not missing) resource, even if the pod is down at the moment, so it can import your reshare, but couldn't if you would have filtered it during export.
